pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -1057,10 +1069,20 @@ func (og *operationGenerator) GenerateUnmapVolumeFunc(
 	volumeToUnmount MountedVolume,
 	actualStateOfWorld ActualStateOfWorldMounterUpdater) (volumetypes.GeneratedOperations, error) {
 
+	// Translate to CSI spec if migration enabled
+	if useCSIPlugin(og.volumePluginMgr, volumeToUnmount.VolumeSpec) {
+		csiSpec, err := translateSpec(volumeToUnmount.VolumeSpec)
+		if err != nil {
+			return volumetypes.GeneratedOperations{}, volumeToUnmount.GenerateErrorDetailed("UnmapVolume.TranslateSpec failed", err)
+		}
+
+		volumeToUnmount.VolumeSpec = csiSpec
+	}
+
 	// Get block volume unmapper plugin
 	var blockVolumeUnmapper volume.BlockVolumeUnmapper
 	blockVolumePlugin, err :=
-		og.volumePluginMgr.FindMapperPluginByName(volumeToUnmount.PluginName)
+		og.volumePluginMgr.FindMapperPluginBySpec(volumeToUnmount.VolumeSpec)
```

Comment:
@msau42 @jingxu97 is the spec always present? Thinking about reconstruction cases here and if `Unmap` is ever called from there without a spec?

Reply:
@mkimuram @wongma7 Have opinions/ideas?

Reply:
```UnmountVolume``` is also called from [```reconciler.cleanupMounts```]( https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/volumemanager/reconciler/reconciler.go#L440), and in this path, ``` volumeToUnmount.VolumeSpec``` is not set. So, current code won’t work well for this path as @davidz627 pointed out.
This path is only called when ```reconstructVolume``` fails, and currently there seems no code for csi migration logic for reconstruction. So, we might be able to just add codes to fall back to ```FindMapperPluginByName``` for ```volumeToUnmount.VolumeSpec``` nil case here as a workaround, and discuss csi migration logic for reconstruction in another PR.

Reply:
Switched back to not use `BySpec` method.